src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -199,9 +202,11 @@ private void prepareRequestForAttachingDocumentToCase(CaseDocumentsMetadata case
 
         for (DocumentHashToken documentHashToken : caseDocumentsMetadata.getDocumentHashTokens()) {
 
-            String hashcodeFromStoredDocument = generateHashToken(UUID.fromString(documentHashToken.getId()));
-            if (!hashcodeFromStoredDocument.equals(documentHashToken.getHashToken())) {
-                throw new ForbiddenException(UUID.fromString(documentHashToken.getId()));
+            if (hashCheckEnabled) {
```

Comment:
Do we need some logging or something here to aid debugging/check if hashCheck is enabled  ?